pkg/server/manager/pipeline.go

```diff
@@ -115,15 +121,29 @@ func (m *pipelineManager) DeletePipeline(projectName string, pipelineName string
 		return err
 	}
 
-	return m.dataStore.DeletePipelineByID(pipeline.ID)
-}
+	// Delete the pipeline records of this pipeline.
+	if err = m.pipelineRecordManager.ClearPipelineRecordsOfPipeline(pipeline.ID); err != nil {
+		logdog.Errorf("Fail to delete all pipeline records of the pipeline %s in the project %s as %s", pipelineName, projectName, err.Error())
+		return err
+	}
 
-// ClearPipelinesOfProject deletes all pipelines in one project.
-func (m *pipelineManager) ClearPipelinesOfProject(projectName string) error {
-	project, err := m.dataStore.FindProjectByName(projectName)
-	if err != nil {
+	if err = m.dataStore.DeletePipelineByID(pipeline.ID); err != nil {
+		logdog.Errorf("Fail to delete the pipeline %s in the project %s as %s", pipelineName, projectName, err.Error())
 		return err
 	}
 
-	return m.dataStore.DeletePipelinesByProjectID(project.ID)
+	return nil
+}
+
+// ClearPipelinesOfProject deletes all pipelines in one project.
+func (m *pipelineManager) ClearPipelinesOfProject(projectID string) error {
```

Comment:
I have concern about this change:
* Changing "projectName" to "projectID" will affect the caller, but the update for caller is not found.
* Where to get the project id? Where to check the existence of this project?

Reply:
now, only the func DeleteProject in pkg/server/manager/project.go call it. and the DeleteProject have checked the existence of this project, I think we don't need to check it twice. or do you have any better suggestion?

Reply:
Use `projectID` is OK.